openslides_backend/presenter/search_users_by_name_email.py

```diff
@@ -0,0 +1,143 @@
+    has_organization_management_level,
+    has_perm,
+)
+from ..permissions.permissions import Permissions
+from ..shared.exceptions import MissingPermission, PresenterException
+from ..shared.filters import FilterOperator, Or
+from ..shared.patterns import Collection, FullQualifiedId
+from ..shared.schema import schema_version
+from .base import BasePresenter
+from .presenter import register_presenter
+
+search_users_by_name_email_schema = fastjsonschema.compile(
+    {
+        "$schema": schema_version,
+        "type": "object",
+        "title": "search_users_by_name_email",
+        "description": "get lists of id, first-, last-name and email for tuples of exact (username, emails)-tuples.",
+        "properties": {
+            "permission_scope": {
+                "type": "array",
+                "prefixItems": [
+                    {
+                        "enum": [1, 2, 3]
+                    },  # Permission scopes Meeting, Committee, Organization
+                    {"type": "number"},  # Id of object, 1 for organization
+                ],
+            },
+            "search": {
+                "type": "array",
+                "minItems": 1,
+                "items": {
+                    "type": "array",
+                    "minItems": 2,
+                    "maxItems": 2,
+                    "prefixItems": [
+                        {"type": "string"},  # username
+                        {"type": "string"},  # email-adress
+                    ],
+                },
+            },
+        },
+        "required": ["permission_scope"],
+        "additionalProperties": False,
+    }
+)
+
+
+@register_presenter("search_users_by_name_email")
+class SearchUsersByNameEmail(BasePresenter):
+    """
+    Collects users with exect usernames or exact emails.
+    """
+
+    schema = search_users_by_name_email_schema
+
+    def get_result(self) -> Any:
+        self.check_permissions(self.data["permission_scope"])
+        result: Dict[str, List[Tuple[int, str, str, str]]] = {}
+        for search in self.data["search"]:
+            if search[0] and search[1]:
+                filter_ = Or(
+                    FilterOperator("username", "=", search[0]),
+                    FilterOperator("email", "=", search[1]),
+                )
+            elif search[0]:
+                filter_ = FilterOperator("username", "=", search[0])
+            elif search[1]:
+                filter_ = FilterOperator("email", "=", search[1])
+            else:
+                continue
+            instances = self.datastore.filter(
+                Collection("user"), filter_, ["id", "first_name", "last_name", "email"]
+            )
+
+            result[f"{search[0]}/{search[1]}"] = [
+                (
+                    instance["id"],
+                    instance.get("first_name", ""),
+                    instance.get("last_name", ""),
+                    instance.get("email", ""),
+                )
```

Comment:
For the result, I woud prefer object notation as well just becase it's more readable. Else the client developer has to look up everytime if email was the second or the last field etc. Also you can then just use the result instances directly without generating the tuple first ;)